internal/pkg/runtime/engines/singularity/process_linux.go

```diff
@@ -145,8 +145,69 @@ func (engine *EngineOperations) checkExec() error {
 	return fmt.Errorf("no %s found inside container", args[0])
 }
 
+// startFuseDrivers starts the required FUSE drivers
+func startFuseDrivers(engine *EngineOperations) error {
+	for _, name := range engine.EngineConfig.GetPluginFuseMounts() {
+		var cfg struct {
+			Fuse struct {
+				DevFuseFd  int
+				MountPoint string
+				Program    []string
+			}
+		}
+
+		if err := engine.EngineConfig.GetPluginConfig(name, &cfg); err != nil {
+			return err
+		}
+
+		fh := os.NewFile(uintptr(cfg.Fuse.DevFuseFd), "fd-"+name)
```

Comment:
There is a chance this could return a `nil` value causing the subsequent deference to panic

Reply:
You are strictly correct: if the filedescriptor is invalid (or a really large value?), it's possible that the various casts inside NewFile produce an error. That's the only path that I could find that could cause this to return nil. I'll add a check for this.